samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/SectionAwareServiceHelperImpl.java

```diff
@@ -114,16 +115,49 @@ public List getGroupReleaseEnrollments(String siteid, String userUid, String pub
 		    log.warn("Unable to find a site with id ("+siteid+") in order to get the enrollments, will return 0 enrollments");
 		}
 
-		for (Iterator eIter = availEnrollments.iterator(); eIter.hasNext(); ) {
-			EnrollmentRecord enr = (EnrollmentRecord)eIter.next();
+		for (EnrollmentRecord enr : availEnrollments) {
 			if (membersInReleaseGroups.contains( enr.getUser().getUserUid())) {
 				enrollments.add(enr);
 			}
 		}
 
 		return enrollments;
 	}
-	
+
+	public List getAllGroupsReleaseEnrollments(String siteid, String userUid, String publishedAssessmentId) {
+		List<EnrollmentRecord> availEnrollments = getAvailableEnrollments(siteid, userUid);
+		List enrollments = new ArrayList();
+
+		HashSet<String> membersInReleaseGroups = new HashSet<>(0);
+		Collection<Group> siteGroups = null;
+		try {
+			Site site = SiteService.getSite(siteid);
+			siteGroups = site.getGroupsWithMember(userUid);
+
+			if (siteGroups.isEmpty() && isUserAbleToGradeAll(siteid, userUid)) {
+				return availEnrollments;
+			}
+
+			Set<String> groupIdsSet = siteGroups.stream()
+				.map(Group::getId)
+				.collect(Collectors.toSet());
+
+			membersInReleaseGroups = new HashSet<>(site.getMembersInGroups(groupIdsSet));
+		}
+		catch (IdUnusedException ex) {
+			// no site found, just log a warning
+			log.warn("Unable to find a site with id ("+siteid+") in order to get the enrollments, will return 0 enrollments");
```

Comment:
```suggestion
			log.warn("Unable to find a site with id ("{}") in order to get the enrollments, will return 0 enrollments", siteid);
```

Reply:
Changed to: 
log.warn("Unable to find a site with id ({}) in order to get the enrollments, will return 0 enrollments", siteid);

Also changed: https://github.com/sakaiproject/sakai/pull/12681/files#diff-b1d3372a6dd0f8da1fede02cdfc1e1251b18ed62f2c223fd585a94c2137befd4R115